### ISSUE TITLE
Ability to change fill color of Progressbar

### ISSUE THREAD

**Issue by ekanna** [fyne-io/fyne#1860]
I want to change fill.color of progressbar but not able to change as applyTheme is a private method.

```go
func (p *progressRenderer) applyTheme() {
	p.background.FillColor = progressBackgroundColor()
	p.bar.FillColor = theme.PrimaryColor()
	p.label.Color = theme.ForegroundColor()
	p.label.TextSize = theme.TextSize()
}
```

Is it possible to make this as a Public Method?

Thanks

**Comment by AlbinoGeek** [comment]
Thanks for the feature request based on our conversation over Slack.

I'm sorry that this widget couldn't be copied out or extended.

Context for those picking this ticket up later:

---

> Koti Kanna
> I saw that private method. I tried to copy paste ProgressBar code but there is an internal package. So could not change it.
> I am getting this err if i copy paste above code `use of internal package fyne.io/fyne/v2/internal/cache not allowed`

---

#1861

**Comment by andydotxyz** [comment]
> Is it possible to make this as a Public Method?

No, we do not expose any renderer internal details in our API. Widget API is based on the behaviour not the render details.
To be able to make the progress bar colour customised we would need to understand the use-case so that the intent can be coded into the API. (i.e. Button.Importance)

**Comment by Jacalz** [comment]
With theme overrides and theming, this should be doable in multiple ways.
